app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/amendment/AmendRegistrationController.scala

```diff
@@ -16,26 +16,248 @@
 
 package uk.gov.hmrc.plasticpackagingtax.registration.controllers.amendment
 
+import play.api.data.Form
 import play.api.i18n.I18nSupport
 import play.api.mvc._
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.plasticpackagingtax.registration.controllers.actions.AuthNoEnrolmentCheckAction
-import uk.gov.hmrc.plasticpackagingtax.registration.models.request.AmendmentJourneyAction
+import uk.gov.hmrc.plasticpackagingtax.registration.forms.contact._
+import uk.gov.hmrc.plasticpackagingtax.registration.models.registration.Registration
+import uk.gov.hmrc.plasticpackagingtax.registration.models.request.{
+  AmendmentJourneyAction,
+  AuthenticatedRequest,
+  JourneyRequest
+}
+import uk.gov.hmrc.plasticpackagingtax.registration.services.CountryService
 import uk.gov.hmrc.plasticpackagingtax.registration.views.html.amendment.amend_registration_page
+import uk.gov.hmrc.plasticpackagingtax.registration.views.html.contact._
+import uk.gov.hmrc.plasticpackagingtax.registration.views.html.partials.amendment.amend_error_page
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 
 import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class AmendRegistrationController @Inject() (
   authenticate: AuthNoEnrolmentCheckAction,
   mcc: MessagesControllerComponents,
   amendmentJourneyAction: AmendmentJourneyAction,
-  amendRegistrationPage: amend_registration_page
-) extends FrontendController(mcc) with I18nSupport {
+  amendRegistrationPage: amend_registration_page,
```

Comment:
It could but I think the additional boilerplate code that would ensue, especially in the tests outweighs any advantages. I'm happy to discuss more widely if you feel strongly about this change?

Reply:
Good call. I'll rename.